src/elm/Page/Community/ObjectiveEditor.elm

```diff
@@ -411,6 +430,39 @@ update msg model loggedIn =
                     |> UR.addExt
                         (Just ClickedSaveObjective |> RequiredAuthentication)
 
+        ClickedArchiveObjetive ->
+            case model.objectiveId of
+                Just objectiveId ->
+                    model
+                        |> UR.init
+                        |> UR.addCmd
+                            (Api.Graphql.mutation loggedIn.shared
+                                (Mutation.completeObjective
+                                    { input = { objectiveId = objectiveId } }
+                                    empty
+                                )
+                                CompletedArchiveMutation
+                            )
+
+                Nothing ->
+                    model
+                        |> UR.init
+
+        CompletedArchiveMutation (Ok _) ->
+            model
+                |> UR.init
+                |> updateObjective msg (\o -> { o | save = Archived })
```

Comment:
since you are redirecting anyway, no need do call this function, nor to have the Archived extra flow